appengine typescript example not working

Hi,
The typescript example is not working out of the box. It shows the error:


```
src/index.ts(3,21): error TS2580: Cannot find name 'process'. Do you need to install type definitions for node? Try npm i --save-dev @types/node.
src/index.ts(14,1): error TS2580: Cannot find name 'module'. Do you need to install type definitions for node? Try npm i --save-dev @types/node.
```

Looking at cloud builder this only causes an error in the build when the NODE_ENV=production

example:

```
--------------------------------------------------------------------------------
Running "node -v"
v12.20.1
Done "node -v" (5.981071ms)
--------------------------------------------------------------------------------
Running "npm ci --quiet (NODE_ENV=development)"

> terminalapp@0.0.1 prepare /workspace
> npm run gcp-build


> terminalapp@0.0.1 gcp-build /workspace
> tsc -p .

added 204 packages in 10.425s
Done "npm ci --quiet (NODE_ENV=development)" (10.892766878s)
--------------------------------------------------------------------------------
Running "cp --archive node_modules /layers/google.nodejs.npm-gcp-build/npm/node_modules"
Done "cp --archive node_modules /layers/google.nodejs.npm-gcp-buil..." (175.265026ms)
--------------------------------------------------------------------------------
Running "npm run gcp-build"

> terminalapp@0.0.1 gcp-build /workspace
> tsc -p .

Done "npm run gcp-build" (5.328688284s)
=== Node.js - NPM (google.nodejs.npm@0.9.0) ===
--------------------------------------------------------------------------------
Running "node -v"
v12.20.1
Done "node -v" (5.197006ms)
DEBUG: Current dependency hash: "87da06bec37af1cb55ec749a04f671c7bb2a79bfda0d0508275ff4ce2998cb22"
DEBUG:   Cache dependency hash: ""
DEBUG: No metadata found from a previous build, skipping cache.
Installing application dependencies.
DEBUG: ***** CACHE MISS: "prod dependencies"
--------------------------------------------------------------------------------
Running "node -v"
v12.20.1
Done "node -v" (4.496962ms)
--------------------------------------------------------------------------------
Running "npm ci --quiet (NODE_ENV=production)"

> terminalapp@0.0.1 prepare /workspace
> npm run gcp-build


> terminalapp@0.0.1 gcp-build /workspace
> tsc -p .

src/index.ts(3,21): error TS2580: Cannot find name 'process'. Do you need to install type definitions for node? Try npm i --save-dev @types/node.
src/index.ts(14,1): error TS2580: Cannot find name 'module'. Do you need to install type definitions for node? Try npm i --save-dev @types/node.
npm ERR! code ELIFECYCLE
npm ERR! errno 2
...
```

the only solution I found was adding a .npmrc file with `production = false` as described in https://github.com/GoogleCloudPlatform/buildpacks/issues/75#issuecomment-686656544

--- a/appengine/typescript/package.json
+++ b/appengine/typescript/package.json
@@ -19,11 +19,11 @@
     "deploy": "gcloud app deploy"
   },
   "dependencies": {
+    "@types/node": "^14.14.7",
     "express": "^4.16.3",
     "typescript": "^4.0.0"
   },
   "devDependencies": {
-    "@types/node": "^14.14.7",
     "mocha": "^8.0.0",
     "wait-port": "^0.2.7",
     "@types/express": "^4.16.0",

